src/cpp/rtps/builtin/discovery/endpoint/EDPSimple.cpp

```diff
@@ -696,7 +696,7 @@ void EDPSimple::assignRemoteEndpoints(const ParticipantProxyData& pdata)
     {
         logInfo(RTPS_EDP,"Adding SEDP Sub Writer to my Sub Reader");
         temp_writer_proxy_data_.guid().entityId = c_EntityId_SEDPSubWriter;
-        temp_writer_proxy_data_.persistence_guid().entityId = c_EntityId_SEDPSubWriter;
+        temp_writer_proxy_data_.persistence_guid().entityId = temp_writer_proxy_data_.persistence_guid().guidPrefix != c_GuidPrefix_Unknown ? c_EntityId_SEDPSubWriter : c_EntityId_Unknown;
```

Comment:
I think `set_persistence_entity_id` should be called here

Reply:
right